explore/circadian-efficacy/data_id_mapping.ipynb

```diff
@@ -0,0 +1,176 @@
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# extract tissue-specific circadian scores by gene\n",
+    "all_genes = circa_db['Entrez.ID'].unique().tolist()\n",
+    "all_tissues = circa_db['tissue'].unique().tolist()\n",
+    "tissue_len = len(all_tissues)\n",
+    "gene_fdr_list = []\n",
+    "gene_amp_list = []\n",
+    "gene_list = []\n",
+    "all_genes_ensg = []\n",
+    "\n",
+    "for gene in all_genes:\n",
+    "    gene_id = circa_db.index[circa_db['Entrez.ID'] == gene]\n",
+    "    if len(gene_id) == tissue_len:\n",
+    "        all_genes_ensg.append(circa_db.iloc[gene_id[0],1])\n",
+    "        gene_list.append(gene)\n",
+    "        \n",
+    "        gene_fdr = np.array(circa_db.iloc[gene_id,4])\n",
+    "        gene_amp = np.array(circa_db.iloc[gene_id,6])\n",
+    "        \n",
+    "        gene_fdr_list.append(gene_fdr)\n",
+    "        gene_amp_list.append(gene_amp)  \n",
+    "\n",
+    "gene_list = np.array(gene_list)\n",
+    "gene_fdr_list = np.array(gene_fdr_list)\n",
+    "gene_amp_list = np.array(gene_amp_list)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# extract tissue-specific expression (median of all samples) by gene\n",
+    "all_gene_exp = mf.map_gtex_expression(all_genes_ensg, all_tissues, list(tissue_uberon.loc[:,'GTEx.name']), list(tissue_uberon.loc[:,'Tissue']), gtex_exp)\n",
+    "all_gene_exp = np.array(all_gene_exp)\n",
+    "all_gene_exp = np.transpose(all_gene_exp)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# combine FDR, amplitude, expression into one ndarray\n",
+    "combine_array = np.concatenate((gene_fdr_list,gene_amp_list,all_gene_exp),axis=1)\n",
+    "\n",
+    "# specify each column name of the ndarray\n",
+    "fdr_names = []\n",
+    "for i in range(0, len(all_tissues)):\n",
+    "    fdr_names.append(all_tissues[i] + '_fdr')\n",
+    "amp_names = []\n",
+    "for i in range(0, len(all_tissues)):\n",
+    "    amp_names.append(all_tissues[i] + '_amp')\n",
+    "exp_names = []\n",
+    "for i in range(0, len(all_tissues)):\n",
+    "    exp_names.append(all_tissues[i] + '_exp')\n",
+    "combine_names = np.concatenate((fdr_names,amp_names,exp_names))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# output ndarray that contains FDR, amplitude, expression of all genes measured\n",
+    "f1 = open('data/circa_db_mapped.tsv', \"w\")\n",
+    "\n",
+    "combine_names_string = '\\t'.join(cn for cn in combine_names)\n",
+    "f1.write('gene_id\\t%s\\n' % combine_names_string)\n",
+    "\n",
+    "for i in range(0, len(combine_array)):\n",
+    "    combine_string = '\\t'.join(format(ca,'.4f') for ca in combine_array[i])\n",
+    "    f1.write('%d\\t%s\\n' % (gene_list[i], combine_string))\n",
```

Comment:
Do not write CSVs / TSVs by manually. Instead use python's `csv` module or create a pandas.DataFrame and use df.to_csv.